packages/editor-ui/src/components/MainHeader/WorkflowDetails.vue

```diff
@@ -587,7 +587,22 @@ export default defineComponent({
 							data: { eventBus: this.eventBus, status },
 						});
 					} catch (error) {
-						this.showError(error, this.$locale.baseText('error'));
+						if ('message' in error) {
+							// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+							switch (error.message) {
+								case 'source_control_not_connected':
+									this.showError(
+										{ ...error, message: '' },
+										this.$locale.baseText('settings.sourceControl.error.not.connected.title'),
+										this.$locale.baseText('settings.sourceControl.error.not.connected.message'),
+									);
+									break;
+								default:
+									this.showError(error, this.$locale.baseText('error'));
+							}
```

Comment:
We can remove the if-else and leave only the switch. The `default` case will handle the situation where `error.message` is undefined.